js/ui/appDisplay.js

```diff
@@ -1519,6 +1519,11 @@ const AppIcon = new Lang.Class({
                 this._updateRunningStyle();
             }));
         this._updateRunningStyle();
+
+        if (app.get_id() === 'com.endlessm.Coding.Chatbox.desktop') {
+            Main.notificationDaemon.gtk.connect('new-gtk-notification-source',
+                                                Lang.bind(this, this._onNewGtkNotificationSource));
+        }
```

Comment:
I'm not entirely sure why this was only done for the chatbox, but that's the way it was in the old code :/

And yes, signal disconnection.